drivers/gpio/gpio_silabs_siwx917.c

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_siwx917_gpio
+
+#include "sl_si91x_driver_gpio.h"
+#include "sl_status.h"
+
+#include <errno.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/irq.h>
+
+/* Zephyr GPIO header must be included after driver, due to symbol conflicts
+ * for GPIO_INPUT and GPIO_OUTPUT between preprocessor macros in the Zephyr
+ * API and struct member register definitions for the SiWx917 device.
+ */
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#if CONFIG_GPIO_SILABS_SIWX917_COMMON_INIT_PRIORITY >= CONFIG_GPIO_INIT_PRIORITY
+#error CONFIG_GPIO_SILABS_SIWX917_COMMON_INIT_PRIORITY must be less than \
+	CONFIG_GPIO_INIT_PRIORITY.
+#endif
+
+#define MAX_PORT_COUNT  4
+#define MAX_PIN_COUNT   16
+#define INVALID_PORT    0xFF
+#define INTERRUPT_COUNT 8
+
+/* Types */
+struct gpio_siwx917_common_config {
+	EGPIO_Type *reg;
+};
+
+struct gpio_siwx917_port_config {
+	/* gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+	const struct device *parent;
+	uint8_t pads[MAX_PIN_COUNT];
+	int port;
+	int hal_port;
+	bool ulp;
+};
+
+struct gpio_siwx917_common_data {
+	/* a list of all ports */
+	const struct device *ports[MAX_PORT_COUNT];
+	sl_gpio_t interrupts[INTERRUPT_COUNT];
+};
+
+struct gpio_siwx917_port_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data common;
+	/* port ISR callback routine address */
+	sys_slist_t callbacks;
+};
+
+/* Functions */
+static int gpio_siwx917_pin_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_siwx917_port_config *cfg = dev->config;
+	const struct device *parent = cfg->parent;
+	const struct gpio_siwx917_common_config *pcfg = parent->config;
+	sl_status_t status;
+	sl_si91x_gpio_driver_disable_state_t disable_state = GPIO_HZ;
+
+	if (flags & GPIO_SINGLE_ENDED) {
+		return -ENOTSUP;
+	}
+
+	uint8_t pad = cfg->pads[pin];
+
+	if (pad != 0xFF && pad != 9) {
+		if (pad == 0) {
+			/* Assign NWP pad to MCU subsystem */
+			status = sl_si91x_gpio_driver_enable_host_pad_selection(
+				(cfg->hal_port << 4) | pin);
+			if (status != SL_STATUS_OK) {
+				return -ENODEV;
+			}
+		} else {
+			/* Assign pad to MCU subsystem */
+			status = sl_si91x_gpio_driver_enable_pad_selection(pad);
+			if (status != SL_STATUS_OK) {
+				return -ENODEV;
+			}
+		}
+	}
```

Comment:
This condition could be written `if (pad == 0) { } else if (pad != 0xFF && pad != 9) { }`. So, you will save one level of indentation.

Reply:
Simplified.